alicloud/resource_alicloud_kvstore_account.go

```diff
@@ -0,0 +1,247 @@
+	if d.HasChange("description") {
+		if err := kvstoreService.WaitForKVstoreAccount(d.Id(), Available, DefaultTimeoutMedium); err != nil {
+			return WrapError(err)
+		}
+		request := r_kvstore.CreateModifyAccountDescriptionRequest()
+		request.RegionId = client.RegionId
+		request.InstanceId = instanceId
+		request.AccountName = accountName
+		request.AccountDescription = d.Get("description").(string)
+
+		raw, err := client.WithRkvClient(func(rkvClient *r_kvstore.Client) (interface{}, error) {
+			return rkvClient.ModifyAccountDescription(request)
+		})
+		if err != nil {
+			return WrapErrorf(err, DefaultErrorMsg, d.Id(), request.GetActionName(), AlibabaCloudSdkGoERROR)
+		}
+		addDebug(request.GetActionName(), raw, request.RpcRequest, request)
+		d.SetPartial("description")
+	}
+
+	if d.HasChange("password") || d.HasChange("kms_encrypted_password") {
+		if err := kvstoreService.WaitForKVstoreAccount(d.Id(), Available, DefaultTimeoutMedium); err != nil {
+			return WrapError(err)
+		}
+		request := r_kvstore.CreateResetAccountPasswordRequest()
+		request.RegionId = client.RegionId
+		request.InstanceId = instanceId
+		request.AccountName = accountName
+
+		password := d.Get("password").(string)
+		kmsPassword := d.Get("kms_encrypted_password").(string)
+
+		if password == "" && kmsPassword == "" {
+			return WrapError(Error("One of the 'password' and 'kms_encrypted_password' should be set."))
+		}
+
+		if password != "" {
+			d.SetPartial("password")
+			request.AccountPassword = password
+		} else {
+			kmsService := KmsService{meta.(*connectivity.AliyunClient)}
+			decryptResp, err := kmsService.Decrypt(kmsPassword, d.Get("kms_encryption_context").(map[string]interface{}))
+			if err != nil {
+				return WrapError(err)
+			}
+			request.AccountPassword = decryptResp.Plaintext
+			d.SetPartial("kms_encrypted_password")
+			d.SetPartial("kms_encryption_context")
+		}
+
+		raw, err := client.WithRkvClient(func(rkvClient *r_kvstore.Client) (interface{}, error) {
+			return rkvClient.ResetAccountPassword(request)
+		})
+		if err != nil {
+			return WrapErrorf(err, DefaultErrorMsg, d.Id(), request.GetActionName(), AlibabaCloudSdkGoERROR)
+		}
+		addDebug(request.GetActionName(), raw, request.RpcRequest, request)
+		d.SetPartial("password")
+	}
+
+	d.Partial(false)
+	return resourceAlicloudKVStoreAccountRead(d, meta)
+}
+
+func resourceAlicloudKVStoreAccountDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	kvstoreService := KvstoreService{client}
+	parts, err := ParseResourceId(d.Id(), 2)
+	if err != nil {
+		return WrapError(err)
+	}
+	request := r_kvstore.CreateDeleteAccountRequest()
+	request.RegionId = client.RegionId
+	request.InstanceId = parts[0]
+	request.AccountName = parts[1]
+
+	raw, err := client.WithRkvClient(func(rkvClient *r_kvstore.Client) (interface{}, error) {
+		return rkvClient.DeleteAccount(request)
+	})
+	if err != nil && !IsExceptedError(err, InvalidAccountNameNotFound) {
```

Comment:
if error code is InvalidAccountNameNotFound, it should return nil directly without call the wait method.

Reply:
done